app/lib/blocs/homework/homework_page_bloc.dart

```diff
@@ -128,7 +128,14 @@ class HomeworkPageBloc extends BlocBase {
 
   /// Take care of closing streams.
   @override
-  void dispose() {
+  Future<void> dispose() async {
+    await Future.wait([
+      _homeworkNotDoneSubject.drain(),
+      _homeworkDoneSubject.drain(),
+      _errorsSubject.drain(),
+      _homeworkListSubject.drain(),
+    ]);
```

Comment:
Needed to `drain` the streams. Otherwise, would be the CI ~50% flaky: https://github.com/SharezoneApp/sharezone-app/runs/6879559070?check_suite_focus=true

```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following StateError was thrown while finalizing the widget tree:
Bad state: You cannot close the subject while items are being added from addStream
When the exception was thrown, this was the stack:
#0      Subject.close (package:rxdart/src/subjects/subject.dart:153:7)
#1      HomeworkPageBloc.dispose (package:sharezone/blocs/homework/homework_page_bloc.dart:137:26)
#2      _BlocProviderState.dispose (package:bloc_provider/src/bloc_provider.dart:53:17)
#3      StatefulElement.unmount (package:flutter/src/widgets/framework.dart:4983:11)
#4      _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1926:13)
#5      _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1924:7)
#6      ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:4807:14)
#7      _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1922:13)
#8      _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1924:7)
#9      ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:4807:14)
#10     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1922:13)
```

Reply:
Let's see if this makes problems. If yes, we need to find a different solution.